### PR TITLE
Graph validations

### LINKED ISSUE #78: No error on refering non-existing process in inport declaration
Example:
```
INPORT=noexist.IN:broken
INPORT=exist.IN:works

exist(foo/Bar)
```
Expected: Error with message explaining that `noexist` referenced in inport `broken` does not exist

### LINKED ISSUE #79: No error on connecting to non-existing processes
Example:
```
right(MyComponent)
'data' -> IN almostright
```
Expected: Error that trying to attach IIP 'data' to non-existing process 'almostright' 

Similar probably happens for port connections

Actual: No error

### LINKED ISSUE #86: No error on missing component name
```
'localhost' -> IN instanceMissingComponentName OUT -> (core/Output)
```

Expected: Error mentioning that process `instanceMissingComponentName` has no component name defined

Actual: No error

### PR DESCRIPTION
**Pull request by bergie**
Validates graph contents against some common errors:

* Fix #78: No error on refering non-existing process in inport declaration
* Fix #79: No error on connecting to non-existing processes
* Fix #86 :No error on missing component name

### CODE REVIEW

File: spec/fbp.coffee [hunk 1]
```diff
@@ -670,3 +670,73 @@ describe 'FBP parser', ->
...
         process: 'Display'
         port: 'out'
+
+  describe 'with FBP string with source node that doesn\'t have a component defined', ->
+    fbpData = """
+    instanceMissingComponentName OUT -> (core/Output)
+    """
+    graphData = null
+    it 'should fail', ->
+      try
```
Comment (jonnor) [inline spec/fbp.coffee]: All these tests should use `chai.expect(() -> ....).to.throw('message')`

File: spec/fbp.coffee [hunk 2]
```diff
@@ -670,3 +670,73 @@ describe 'FBP parser', ->
...
+    OUTPORT=noexist.OUT:broken
+    exist(foo/Bar)
+    """
+    graphData = null
+    it 'should fail', ->
+      try
+        graphData = parser.parse fbpData, caseSensitive:true
+      catch e
+        chai.expect(e).to.be.an 'error'
+        chai.expect(e.message).to.contain 'connected to an undefined source node'
```
Comment (jonnor) [inline spec/fbp.coffee]: Error messages should also contain name of instance with the problem?

File: spec/fbp.coffee (hunk 1 above)
Comment (jonnor) [reply spec/fbp.coffee]: Sorry, is `chai.expect(() -> ....).to.throw(ErrorType, 'message')`

**Comment by bergie** [comment]
Due to the generated code being in this repo, I'll make the requested changes in #88 to prevent rebasing issues.

**Pull request merged**